packages/react/src/components/FileUploader/index.js

```diff
@@ -0,0 +1,153 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+
+import Icon from "../Icon";
+import Spinner from "../Spinner";
+import Text from "../Text";
+import UploaderItem from "../UploaderItem";
+
+import { ICONS } from "../../constants";
+
+const defaultTranslate = ({ defaultText }) => defaultText;
+
+const texts = {
+  add_document: {
+    id: "flamingo-file-uploader-add-document",
+    defaultText: "Add a document",
+  },
+  error: {
+    id: "flamingo-file-uploader-error-title",
+    defaultText: "Upload failed",
+  },
+  try_again: {
+    id: "flamingo-file-uploader-error-try-again",
+    defaultText: "Click here to try again",
+  },
+};
+
+const FileUploader = ({
+  children,
+  className,
+  file,
+  name,
+  onChange,
+  translate,
+  isLoading,
+  isErrored,
+  overrides,
+  ...props
+}) => {
+  const [innerFile, setInnerFile] = React.useState(undefined);
+  const hasFile = !!file || !!innerFile;
+
+  const handleFileChange = e => {
+    const [eventFile] = e.target.files;
+
+    setInnerFile(eventFile);
+    onChange(eventFile);
+  };
+
+  const handleResetState = () => {
+    setInnerFile(undefined);
+  };
+
+  React.useEffect(
+    () => {
+      setInnerFile(file);
+    },
+    [file]
+  );
+
+  return (
+    <div
+      className={cx("FileUploader", {
+        [className]: className,
```

Comment:
I think we should keep this flexibility. If someone wants to update the styles, so be it.
It can also be useful for us.

Let's say you want to extend the `FileUploader` component, you can do it by this quick one-liner `const MyFileUploader = props => <FileUploader className="FileUploader--custom" {...props} />`